### ISSUE TITLE
Make django-allauth a selectable option

### ISSUE THREAD

**Issue by Tamerz** [pydanny/cookiecutter-django#2969]
## Description

It would be great if django-allauth became an option instead of a default.

## Rationale

A lot of sites I create are for internal users and don't need all the features that come with django-allauth. It is not trivial removing all the references from it in my projects without breaking something.

**Comment by 9mido** [comment]
I agree. It would be nice if there was a tutorial (videos preferably) that covered the django authentication framework more in depth so that users do not have to rely on simply installing allauth. Or having code snippets that cover specific features of allauth so that users could choose whatever features they want for their authentication.  

Since django-allauth is maintained by one person that is in itself a concern. Improvements that some see as important get sidelined sometimes for years or just forgotten about while non-important things like social features get merged without blinking an eye. Then supposedly you need to pay the creator to get something merged. That is not truly open source. At any moment, the owner could die, pull something like Red Hat did with CentOS, or no longer support it leaving us all screwed unless it gets forked by somebody else. In all honesty, allauth should become a jazzband project and given to the community to take care of. Leaving it up to one person to decide the fate of allauth creates problems for future development efforts.

**Comment by omarsumadi** [comment]
Huge upvote!!

**Comment by rustanacexd** [comment]
Anyone? :D

**Comment by browniebroke** [comment]
In a month time, there seems to be some interest.

Trying to evaluate how difficult such an option would be to maintain... Doing a quick search in the code, it doesn't seem too difficult, however, I see in `base.html` that several views are currently provided by allauth:

https://github.com/pydanny/cookiecutter-django/blob/bf2d5370be5334f7037aead9f544aed8d5416cf2/%7B%7Bcookiecutter.project_slug%7D%7D/%7B%7Bcookiecutter.project_slug%7D%7D/templates/base.html#L54-L72

Should the template provide these if allauth is opted-out? If so, I guess we would need to have another `account` app that we would delete in post gen hook if allauth is opted-in...

**Comment by Andrew-Chen-Wang** [comment]
Imo if you don't need allauth, then I don't find it necessary for cookiecutter-django to create its own account app to make up for the registration. I think people are simply asking to not have a signup and stuff in the first place. So having the option to remove django-allauth is basically asking cookiecutter django to not have registration integrated with the UI.

**Comment by areski** [comment]
@Andrew-Chen-Wang I think there is also the scenario where you want to use something else, specially if you are not interested by the social apps, for instance I like to use https://github.com/macropin/django-registration in my Django projects.
My 2 cents :)

**Comment by femimada** [comment]
Upvoting this as well, spent a lot of time trying to remove this the other day

**Comment by femimada** [comment]
Upvoting this as well.  Allauth is a great package but it's a nightmare to remove from existing project

**Comment by denny-sam** [comment]
Hi, is this feature up for consideration? I can start working on it if it is. Thanks!
